ui.tests/test-module/specs/textinput/textinput.authoring.spec.js

```diff
@@ -136,5 +144,20 @@ describe('Page - Authoring', function () {
       testTextInputBehaviour(textInputEditPathSelector, textInputDrop, true);
     });
 
+    it('Test z-index of Rule editor iframe', function () {
+      dropTextInputInSites();
+      cy.openSidePanelTab("Content Tree");
+      cy.openEditableToolbar(sitesSelectors.overlays.overlay.component + textInputEditPathSelector);
+      cy.invokeEditableAction("[data-action='editexpression']");
+      cy.get("#af-rule-editor").should("be.visible");
+      cy.get("#af-rule-editor")
+          .invoke("css", "z-index")
+          .should("equal", '10');
+      getRuleEditorIframe().find("#objectNavigationTree").should("be.visible");
+      getRuleEditorIframe().find("#create-rule-button").should("be.visible");
+      cy.wait(1000)
```

Comment:
we should not use wait. we already have support of promises. eg: https://github.com/adobe/aem-core-forms-components/blob/master/ui.tests/test-module/libs/support/commands.js#L251